src/main/java/com/iexec/common/utils/BytesUtils.java

```diff
@@ -44,9 +46,43 @@ public static String hexStringToAscii(String hexString) {
     }
 
     public static boolean isHexaString(String hexaString) {
-        return Numeric.cleanHexPrefix(hexaString).matches("\\p{XDigit}+"); // \\p{XDigit} matches any hexadecimal character
+        return StringUtils.isNotEmpty(hexaString)
+                && Numeric.cleanHexPrefix(hexaString).matches("\\p{XDigit}+"); // \\p{XDigit} matches any hexadecimal character
     }
 
+    /**
+     * Validates hexadecimal string input and verifies its length.
+     * Hexadecimal string input must contain 0x prefix.
+     *
+     * @param hexString        hexadecimal input
+     * @param expectedByteSize expected byte size
+     * @return true if
+     */
+    public static boolean isHexStringWithProperBytesSize(String hexString,
+                                                        int expectedByteSize) {
+        return expectedByteSize > 0
+                && StringUtils.isNotEmpty(hexString)
+                && Numeric.containsHexPrefix(hexString)
+                && isHexaString(hexString)
+                && Numeric.hexStringToByteArray(hexString).length == new byte[expectedByteSize].length;
```

Comment:
Why not?
```suggestion
                && Numeric.hexStringToByteArray(hexString).length == expectedByteSize;
```

Reply:
8 bits = 1 byte -> 2 char slots (0xff fits in 1 byte)?

Reply:
Not sure about what you mean, but `new byte[expectedByteSize].length` is strictly equal to `expectedByteSize` as `new type[n]` creates an array of length `n` and `array.length`retrieves the length of the array...

Reply:
"new byte[expectedByteSize].length is strictly equal to expectedByteSize"
totally agree, let me update that :+1: